hazelcast/src/main/java/com/hazelcast/internal/config/ConfigDataSerializerHook.java

```diff
@@ -230,6 +236,9 @@ public DataSerializableFactory createFactory() {
         constructors[PERMISSION_CONFIG] = arg -> new PermissionConfig();
         constructors[BITMAP_INDEX_OPTIONS] = arg -> new BitmapIndexOptions();
         constructors[DATA_PERSISTENCE_CONFIG] = arg -> new DataPersistenceConfig();
+        constructors[TIERED_STORE_CONFIG] = arg -> new TieredStoreConfig();
+        constructors[MEMORY_TIER_CONFIG] = arg -> new MemoryTierConfig();
+        constructors[DISK_TIER_CONFIG] = arg -> new DiskTierConfig();
```

Comment:
> No, we don't reference device config from tieredStoreConfig or mapConfig, since device config is more global and local to a given HZ instance.

since `DeviceConfig` is not shared among HZ members, we don't need it to be serializable, no?

Reply:
Ah, you're right :man_facepalming: